ops/framework.py

```diff
@@ -827,6 +831,32 @@ def _next_event_key(self) -> str:
         self._stored['event_count'] += 1
         return str(self._stored['event_count'])
 
+    def _event_is_in_storage(
+        self, observer_path: str, method_name: str, event_path: str, event_data: Dict[str, Any]
+    ) -> bool:
+        """Check if there is already a notice with the same snapshot in the storage."""
+        # Check all the notices to see if there is one that is the same other
+        # than the event ID.
+        for (
+            existing_event_path,
+            existing_observer_path,
+            existing_method_name,
+        ) in self._storage.notices():
+            if (
+                existing_observer_path != observer_path
+                or existing_method_name != method_name
```

Comment:
I wonder if a convenience method on `Handle` would be better to isolate the path format logic.

Reply:
I wondered about that too, but thought that it was better to keep this change more isolated, since I'm not sure that "compare paths ignoring keys" has any more generalised use. I could move some of this there if that was preferred.